demo_helper.py

```diff
@@ -0,0 +1,155 @@
+
+    # so delta_t is given in ns
+
+    orig_energy = np.sum(orig_signal_upsampled**2)
+    test_energy = np.sum(test_signal_upsampled**2)
+    energy_rel_diff = (test_energy - orig_energy) / orig_energy
+
+    return (CC_zeroshift, CC_optimized_timeshift, delta_t, energy_rel_diff)
+
+
+
+def plot_pulse_and_spectrum(orig_pulse, interpolated_pulse, x, y, cutoff_freq, pol):
+    """
+    Plots an interpolated pulse together with a 'true' simulated pulse
+
+    Parameters
+    ----------
+    orig_pulse : time series, 1D array
+    interpolated_pulse : idem
+    x : the x position (float), for annotation in the plot
+    y : idem for y
+    cutoff_freq : value of estimated cutoff frequency, for annotation only
+    pol : polarization number
+    """
+    time_axis = 0.1 * np.arange(len(orig_pulse))
+    radius = np.sqrt(x**2 + y**2)
+    freqs = get_freq_axis(orig_pulse)
+
+    (CC_zeroshift, CC_optimized_timeshift, delta_t, energy_rel_diff) = get_crosscorrelation(orig_pulse, interpolated_pulse)
+
+    fig, ax = plt.subplots(figsize=(10.67, 4), nrows=1, ncols=2)
+    ax1, ax2 = ax[0], ax[1]
+
+    #plt.figure()
+    ax1.plot(time_axis, 1.0e6 * orig_pulse, label='orig pulse', lw=2)
+    ax1.plot(time_axis, 1.0e6 * interpolated_pulse, label='interpolated pulse', lw=2)
+    residual = 1.0e6 * (interpolated_pulse - orig_pulse)
+    ax1.plot(time_axis, residual, label='difference', lw=2, c='g')
+
+    interpolated_energy = np.sum(interpolated_pulse**2)
+    orig_energy = np.sum(orig_pulse**2)
+    #fixed_pulse = interpolated_pulse * np.sqrt(orig_energy / interpolated_energy)
+    #plt.plot(time_axis, fixed_pulse, label='fixed pulse', c='r', lw=1)
+    ax1.grid()
+    ax1.set_xlabel('Time [ ns ]')
+    ax1.set_ylabel(r'E-field [ $\mu$V/m ]')
+    ax1.set_xlim(0.0, 50.0)
+    ax1.legend(loc='best')
+
+    #plt.figure()
+    orig_pulse_powerspec = np.abs(np.fft.rfft(orig_pulse))**2
+    interp_pulse_powerspec = np.abs(np.fft.rfft(interpolated_pulse))**2
+    ax2.plot(freqs, orig_pulse_powerspec, label='Orig pulse')
+    ax2.plot(freqs, interp_pulse_powerspec, label='Interpolated pulse')
+    ax2.text(0.98, 0.40, 'Position x = %3.1f, y = %3.1f, r = %3.2f m, pol = %d' % (x, y, radius, pol), transform=plt.gca().transAxes, ha='right') #, va='right')
+    ax2.text(0.98, 0.30, 'CC = %1.5f, CC_max = %1.5f' % (CC_zeroshift, CC_optimized_timeshift), transform=plt.gca().transAxes, ha='right')
+    ax2.text(0.98, 0.20, 'delta_t = %1.2f ns, cutoff freq = %3.1f MHz' % (delta_t, cutoff_freq), transform=plt.gca().transAxes, ha='right')
+
+    #plt.yscale('log')
+    ax2.grid()
+    ax2.set_xlabel('Frequency [ MHz ]')
+    ax2.set_ylabel('Power spectrum [ a.u. ]')
+    ax2.set_xlim(0, 500)
+    ax2.set_ylim(0.0, 1.2*np.max(orig_pulse_powerspec))
+    ax2.legend(loc='best')
+
+
+def read_data_hdf5(filename):
+    """
+    Reading in the demo data hdf5 file.
```

Comment:
Is this a 'standard' CoREAS output? If so, consider saying so.

Reply:
Yes, converted to on-sky. Done.